test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs

```diff
@@ -1655,12 +1710,27 @@ public async Task GetTrustResultAsync_WithRevokedPrimaryCertificate_ReturnsSuspe
                 var issueCertificateOptions = IssueCertificateOptions.CreateDefaultForEndCertificate();
                 var bcCertificate = certificateAuthority.IssueCertificate(issueCertificateOptions);
                 var timestampService = await _fixture.GetDefaultTrustedTimestampServiceAsync();
+#if IS_DESKTOP
+                var privateKey = DotNetUtilities.ToRSA(issueCertificateOptions.KeyPair.Private as RsaPrivateCrtKeyParameters);
+#else
+                var rsaParameters = DotNetUtilities.ToRSAParameters(issueCertificateOptions.KeyPair.Private as RsaPrivateCrtKeyParameters);
+
+                RSACryptoServiceProvider rsaCsp = new RSACryptoServiceProvider();
 
+                rsaCsp.ImportParameters(rsaParameters);
+
+                var privateKey = rsaCsp;
+#endif
+#if IS_DESKTOP
```

Comment:
Consider checking that the type casting is successful. Also, consider minimizing the repeated code in `#if`, `#else` directives. For example:

```suggestion
                var privateKeyPair = issueCertificateOptions.KeyPair.Private as RsaPrivateCrtKeyParameters;
                Assert.NotNull(privateKeyPair);
#if IS_DESKTOP
                var privateKey = DotNetUtilities.ToRSA(privateKeyPair);
#else
                var rsaParameters = DotNetUtilities.ToRSAParameters(privateKeyPair);
                RSACryptoServiceProvider rsaCsp = new RSACryptoServiceProvider();
                rsaCsp.ImportParameters(rsaParameters);
                var privateKey = rsaCsp;
#endif
```

Reply:
Thanks! Fixed the type casting by:
```
Assert.IsType<RsaPrivateCrtKeyParameters>(keyPair.Private);

var privateKeyParameters = (RsaPrivateCrtKeyParameters)keyPair.Private;
```